tests/factory/test_factory_invalid_options.py

```diff
@@ -1,187 +1,42 @@
-import copy
```

Comment:
The file name should contain `missing_mandatory_options` to mimic the former class name.

Reply:
renamed.